qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1898,6 +1898,64 @@ def count_ops_longest_path(self):
                 op_dict[name] += 1
         return op_dict
 
+    def _get_qubit_input_output_node(self, qubit, in_or_out=False):
+        """Returns qubit and input or output node from an index."""
+        nodes = self.output_map if in_or_out else self.input_map
+        # Check if index is passed
+        if isinstance(qubit, int):
+            if qubit >= self.num_qubits():
+                raise DAGCircuitError(f"Qubit index {qubit} is out of range")
+            qubit = list(nodes.keys())[qubit]
+        if qubit not in self.qubits:
+            raise DAGCircuitError("Qubit was not found in circuit")
+        return (qubit, nodes.get(qubit, None))
+
+    def get_qubit_input_node(self, qubit):
+        """Returns qubit and input node from a qubit index."""
+        return self._get_qubit_input_output_node(qubit)
+
+    def get_qubit_output_node(self, qubit):
+        """Returns qubit and output node from a qubit index."""
+        return self._get_qubit_input_output_node(qubit, True)
+
+    def get_causal_cone(self, qubit_index):
+        """Returns causal cone of a qubit."""
```

Comment:
One thing that is important for a new public method on `DAGCircuit` is to ensure the docstring documents the behavior of the function, why you would use it, the input arguments and their types, and the output type as well. This docstring will be part of the documentation that gets hosted for the `DAGCircuit` after it's merged and released. So could you expand a bit more here about what a causal cone is and why/when you would need to get it. Similarly, the input args and the return should be documented so users can read the documentation and know how to interface with the method.

Reply:
Got it, will do this promptly.